server/insertlist.go

```diff
@@ -0,0 +1,135 @@
+package server
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/TheYeung1/yata-server/database"
+	"github.com/TheYeung1/yata-server/model"
+	log "github.com/sirupsen/logrus"
+)
+
+type InsertListInput struct {
+	ListID string
+	Title  string
+}
+
+// Validate returns an error if the input does not pass validation.
+func (input *InsertListInput) Validate() error {
+	if len(input.ListID) == 0 {
+		return errors.New("ListID cannot be empty")
+	}
+	if len(input.ListID) > 100 {
+		return errors.New("ListID length cannot exceed 100 characters")
+	}
+	if len(input.ListID) != len(strings.TrimSpace(input.ListID)) {
+		return errors.New("ListID cannot be prefixed or suffixed with spaces")
+	}
+	if len(input.Title) == 0 {
+		return errors.New("Title cannot be empty")
+	}
+	if len(input.Title) > 100 {
+		return errors.New("Title length cannot exceed 100 characters")
+	}
+	if len(input.Title) != len(strings.TrimSpace(input.Title)) {
+		return errors.New("Title cannot be prefixed or suffixed with spaces")
+	}
+	return nil
+}
+
+type InsertListOutput struct {
+	ListID string
+}
+
+func (s *Server) InsertList(w http.ResponseWriter, r *http.Request) {
+	uid, err := getUserIDFromContext(r)
+	if err != nil {
+		log.WithError(err).Error("failed to get user ID from request context")
+		renderInternalServerError(w)
+		return
+	}
+	log.WithField("userID", uid).Debug("insert list called")
+
+	var input InsertListInput
+	if err := bindJSON(r.Body, &input); err != nil {
+		log.WithError(err).Info("failed to bind input")
+		renderBadRequest(w, "malformed input")
+		return
+	}
+	log.WithField("input", input).Debug("input bound")
+
+	if err := input.Validate(); err != nil {
+		log.WithError(err).Info("failed to normalize and validate input")
+		renderBadRequest(w, err.Error())
+		return
+	}
+
+	yl := model.YataList{
+		UserID: model.UserID(uid[0]),
```

Comment:
Question: why the `[0]`? I don't understand why we only want the first character of the uid...

Reply:
Cool. Will fix.